### PR TITLE
[Doc] Migrate WebDoc to doxygen 1.9.4 and improve layout

### PR DESCRIPTION
**Pull request by benj5378**
- Migrate BuildWebDoc.cfg.in to doxygen 1.9.4
- Clean up footer and header for WebDoc

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7192">pipeline status</a> for feature branch [PR_7192](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7192). Pipeline [589079989](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589079989) was triggered at [99c9cd9](https://github.com/FreeCAD/FreeCAD/pull/7192/commits/99c9cd9bca124627c899b0b1b2f29e4b9ab4357e). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
Thank you for this awesome contribution! Can you attach a screenshot of the cosmetic changes it introduces? (background color etc...)

**Comment by benj5378** [comment]
Thanks for the compliments @luzpaz. It is completely similar to the current on https://freecad.github.io/SourceDoc/modules.html, except that the footer is a little taller, there is no CC logo (I think it was previously a part of doxygen, but now removed).

It's mostly under the hood changes. There was a lot of elements with bootstrap classes even though bootstrap weren't included. 2 CSS files overwriting one another - one holding a lot of CSS not relevant to the documentation.

**Comment by yorikvanhavre** [comment]
Awesome!! Happy someone gave some love to the doxygen docs ;) My wish is some day be able to integrate with https://github.com/FreeCAD/FreeCAD-documentation ...
Thanks!

File: src/Doc/BuildWebDoc.cfg.in [hunk 1]
```diff
@@ -1,602 +1,929 @@
...
-# sources only. Doxygen will then generate output that is more tailored for
-# Java. For instance, namespaces will be presented as packages, qualified
-# scopes will look different, etc.
+# Set the OPTIMIZE_OUTPUT_JAVA tag to YES if your project consists of Java or
+# Python sources only. Doxygen will then generate output that is more tailored
+# for that language. For instance, namespaces will be presented as packages,
+# qualified scopes will look different, etc.
+# The default value is: NO.
 
 OPTIMIZE_OUTPUT_JAVA   = YES
```
Comment (benj5378) [inline src/Doc/BuildWebDoc.cfg.in:291]: @yorikvanhavre I was wondering if this was on purpose? Optimiza for... Java?

Comment (yorikvanhavre) [reply src/Doc/BuildWebDoc.cfg.in:291]: Probably some default setting that was still turned on without us noticing... Indeed it definitely should be NO I think

**Pull request merged**